go/mysql/datetime/datetime.go

```diff
@@ -569,8 +569,12 @@ func (dt DateTime) Round(p int) (r DateTime) {
 	return r
 }
 
-func (dt DateTime) toSeconds() int {
-	return (dt.Date.Day()-1)*secondsPerDay + dt.Time.toSeconds()
+func (dt DateTime) toDuration() time.Duration {
+	dur := dt.Time.toDuration()
+	if !dt.Date.IsZero() {
```

Comment:
When we have a zero date (and thus we're a Time type), we don't include the day bit to have a correct duration.